### PR TITLE
 table edit default behavior

### LINKED ISSUE #557: Default behaviors of edit, add, and delete for tables
**Description**
The goal would be to have default behaviors for _add_, _delete_, and _edit_ features of tables to facilitate their use. This way, a user wouldn't be forced to create specific functions (_on_add_, _on_delete_, and _on_edit_). In most cases, these functions could manipulate the data as wanted. If the user wants a specific behavior, they could create their specific functions as it is done now.

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change
- [ ] Ensure any change is well documented

### LINKED ISSUE #1488: [🐛 BUG] Coverage report is failing
### What went wrong? 🤔

The cover report does not have the correct authorization to update the PR and failed.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [X] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #557
resolves #1488 

```
import pandas as pd

from taipy.gui import Gui


def get_table():
    dic = {"key": [1,2,3], "Value":[3,4,5]}
    return pd.DataFrame(dic)

table = get_table()
page_content = """
<|{table}|table|show_all|editable|not on_add|>
"""

pages = {"content": page_content}

gui = Gui(pages=pages)
gui.run(title="557 Table default edit")

```

### CODE REVIEW

File: taipy/gui/_renderers/factory.py [hunk 1]
```diff
@@ -509,10 +509,10 @@ class _Factory:
                 ("width", PropertyType.string_or_number, "100%"),
                 ("height", PropertyType.string_or_number, "80vh"),
                 ("active", PropertyType.dynamic_boolean, True),
-                ("editable", PropertyType.dynamic_boolean, True),
-                ("on_edit", PropertyType.function),
-                ("on_delete", PropertyType.function),
-                ("on_add", PropertyType.function),
+                ("editable", PropertyType.dynamic_boolean, False),
+                ("on_edit", PropertyType.function, gui._get_call_method_name('_table_on_edit')),
```
Comment (FabienLelaquais) [inline taipy/gui/_renderers/factory.py]:
```suggestion
                ("on_edit", PropertyType.function, gui._get_call_method_name("_table_on_edit")),
```

File: taipy/gui/_renderers/factory.py [hunk 2]
```diff
@@ -509,10 +509,10 @@ class _Factory:
...
+                ("on_delete", PropertyType.function, gui._get_call_method_name('_table_on_delete')),
```
Comment (FabienLelaquais) [inline taipy/gui/_renderers/factory.py]:
```suggestion
                ("on_delete", PropertyType.function, gui._get_call_method_name("_table_on_delete")),
```

File: taipy/gui/_renderers/factory.py [hunk 3]
```diff
@@ -509,10 +509,10 @@ class _Factory:
...
+                ("on_add", PropertyType.function, gui._get_call_method_name('_table_on_add')),
```
Comment (FabienLelaquais) [inline taipy/gui/_renderers/factory.py]:
```suggestion
                ("on_add", PropertyType.function, gui._get_call_method_name("_table_on_add")),
```

File: taipy/gui/data/pandas_data_accessor.py [hunk 4]
```diff
@@ -453,3 +454,32 @@ def get_data(
...
+    def _on_delete(self, df: pd.DataFrame, payload: t.Dict[str, t.Any]):
+        return df.drop(payload["index"])
+
+    def _on_add(self, df: pd.DataFrame, payload: t.Dict[str, t.Any]):
+        # Save the insertion index
+        index = payload["index"]
+        # Create the new row (Column value types must match the original DataFrame's)
+        col_types = self.get_col_types("", df)
+        if col_types:
+            new_row = [0 if is_numeric_dtype(df[c]) else "" for c in list(col_types)]
```
Comment (FabienLelaquais) [inline taipy/gui/data/pandas_data_accessor.py]: We must think about providing a 'default new row content'

File: .github/workflows/frontend.yml [hunk 5]
```diff
@@ -70,10 +74,11 @@ jobs:
...
         if: matrix.os == 'ubuntu-latest' && github.event_name == 'pull_request' && steps.cache-gui-fe-build.outputs.cache-hit != 'true'
-        uses: artiomtr/jest-coverage-report-action@v2.3.0
+        uses: artiomtr/jest-coverage-report-action@v2
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           threshold: "80"
           working-directory: "./frontend/taipy-gui"
           skip-step: install
           annotations: failed-tests
+          # base-coverage-file: ./report.json
```
Comment (FabienLelaquais) [inline .github/workflows/frontend.yml:84]: Don't you want to drop that?

File: .github/workflows/frontend.yml [hunk 6]
```diff
@@ -2,14 +2,18 @@ name: Frontend Tests & Coverage
...
-    branches: [ develop ]
+    branches: [develop]
     paths:
-        - 'frontend/taipy-gui/**'
+      - "frontend/taipy-gui/**"
   workflow_dispatch:
 
 permissions:
+  # Required to checkout the code
+  # contents: write
```
Comment (FabienLelaquais) [inline .github/workflows/frontend.yml]: Remove?

**Review by FabienLelaquais — DISMISSED** [review]
👍

File: .github/workflows/frontend.yml (hunk 5 above)
Comment (FredLL-Avaiga) [reply .github/workflows/frontend.yml:84]: I'd rather keep it for understanding

**Review by FabienLelaquais — APPROVED** [review]
👌

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

**Review by namnguyen20999 — APPROVED** [review]
LGTM

**Comment by bobbyshermi** [comment]
Thank you for implementing this. That's great.

A few questions, based on the example you gave: 
`<|{table}|table|show_all|editable|not on_add|>`

1. By default, the table is not editable, right?
2. What is the `not on_add` property? What is its default behavior and the default values for each column?

Thanx

**Comment by FredLL-Avaiga** [comment]
Replies in the issue #557

**Pull request merged**